In [2]:
import pandas as pd
import numpy as np

In [3]:
pd.options.display.max_columns = 999

In [4]:
df = pd.read_csv('arquivos/dadosalunos-21-08-2023.csv')

# Informações Iniciais

In [5]:
df.head()

ANO_OCORRENCIA    CD_ALUNO  CD_CURSO  CD_ORIGEM_ALUNO CD_TURNO  \
0            2014   145457572         3                1        N   
1            2015   125569920        15                0        N   
2            2018   131671183        34                1        V   
3            2011  1236669952         6                4        N   
4            2009  1748606897         3                5        N   

  DE_NATURALIDADE DH_MATRICULA DH_SITUACAO  DT_ANO_INGRES DT_NASC_ALUNO  \
0      BRASILEIRO   2014-01-10  2014-08-26           2014    1974-07-08   
1      BRASILEIRO   2015-01-24  2016-03-24           2015    1971-02-08   
2      BRASILEIRO   2015-01-13  2019-01-18           2015    1971-07-12   
3      BRASILEIRO   2010-02-01  2011-07-19           2010    1974-05-25   
4      BRASILEIRA   2004-02-01  2009-01-30           2004    1966-07-27   

   LT_ESTADO_CIVIL LT_MODALIDADE LT_NACIONALIDADE  LT_RACA NM_CIDADE_AULA  \
0              1.0             P                B        4        MARINGA   
1              2.0             P                B        2        MARINGA   
2              1.0             P                B        5        MARINGA   
3              1.0             P                B        0    PIRAPOZINHO   
4              2.0             P                B        0        MARINGA   

  NM_CIDADE_EST2GRAU NM_CIDADE_FERIAS       NM_CIDADE_NASC NM_PAIS_NASC  \
0            MARINGA          MARINGA              MARINGA       BRASIL   
1                NaN              NaN           MANDAGUARI       BRASIL   
2            MARINGA          MARINGA              MARINGA       BRASIL   
3                NaN              NaN  PRESIDENTE PRUDENTE       BRASIL   
4                NaN          MARINGA                  NaN       BRASIL   

  NU_SEMESTRE2_VEST SG_SEXO_ALUNO SG_UF_AULA SG_UF_EST2GRAU SG_UF_FERIAS  \
0                 I             M         PR             PR           PR   
1               NaN             M         PR            NaN          NaN   
2                 V             M         PR             PR           PR   
3               NaN             M         SP            NaN          NaN   
4               NaN             F         PR            NaN           PR   

  SG_UF_NASC_ALUNO ST_ALUNO TP_DEFIC_INEP TP_ESCOLA  TP_INGRES_CURSO  \
0               PR        2      NNNNNNNN        GI                1   
1               PR        7      NNNNNNNN       NaN                6   
2               PR        7      NNNNNNNN        GI                1   
3               SP        8           NaN         E                6   
4               RS        9           NaN        GI               11   

  TP_OPCAO_COTAS  
0              N  
1            NaN  
2              N  
3              N  
4              N

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71221 entries, 0 to 71220
Data columns (total 30 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   ANO_OCORRENCIA      71221 non-null  int64  
 1   CD_ALUNO            71221 non-null  int64  
 2   CD_CURSO            71221 non-null  int64  
 3   CD_ORIGEM_ALUNO     71221 non-null  int64  
 4   CD_TURNO            71221 non-null  object 
 5   DE_NATURALIDADE     50710 non-null  object 
 6   DH_MATRICULA        71221 non-null  object 
 7   DH_SITUACAO         71221 non-null  object 
 8   DT_ANO_INGRES       71221 non-null  int64  
 9   DT_NASC_ALUNO       71221 non-null  object 
 10  LT_ESTADO_CIVIL     70989 non-null  float64
 11  LT_MODALIDADE       71221 non-null  object 
 12  LT_NACIONALIDADE    70924 non-null  object 
 13  LT_RACA             71221 non-null  int64  
 14  NM_CIDADE_AULA      70245 non-null  object 
 15  NM_CIDADE_EST2GRAU  56560 non-null  object 
 16  NM_C

# Análise Colunas

### ANO_OCORRENCIA

Ano letivo. A ocorrência é em casos mais específicos, como formado, transferido e cancelado. Então não é o ano da última 

In [7]:
df['ANO_OCORRENCIA'].nunique(dropna=False)

23

In [8]:
np.sort(df['ANO_OCORRENCIA'].unique())

array([   0, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011,
       2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022,
       2023], dtype=int64)

In [9]:
df['ANO_OCORRENCIA'].value_counts(dropna=False).sort_index()

ANO_OCORRENCIA
0       7774
2002      71
2003     427
2004     732
2005    1573
2006    2837
2007    2729
2008    4149
2009    4239
2010    3781
2011    3675
2012    3570
2013    3813
2014    3599
2015    3636
2016    3178
2017    2972
2018    3133
2019    3248
2020    3279
2021    3683
2022    3666
2023    1457
Name: count, dtype: int64

### CD_ALUNO

In [10]:
df['CD_ALUNO'].nunique()

71221

### CD_CURSO

In [11]:
df['CD_CURSO'].nunique(dropna=False)

45

In [12]:
np.sort(df['CD_CURSO'].unique())

array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,
       19, 21, 22, 23, 24, 25, 26, 27, 28, 30, 31, 33, 34, 35, 36, 37, 38,
       39, 40, 42, 43, 44, 45, 46, 47, 48, 49, 50], dtype=int64)

**Ausência dos cursos 18, 20, 29, 32 e 41**

In [13]:
df['CD_CURSO'].value_counts(dropna=False).sort_index()

CD_CURSO
1     2814
2     3842
3     3518
4     2319
5     1901
6     2035
7     2050
8     3706
9     1961
10    2157
11    3104
12    2694
13    2940
14    2086
15    1429
16    1801
17    1799
19    2136
21    1969
22     970
23     869
24     965
25     997
26    1047
27     660
28     393
30    1035
31     993
33    1178
34     859
35    1026
36     953
37    2825
38     952
39     808
40     909
42     521
43     853
44     870
45     688
46     978
47    1089
48     971
49     731
50     820
Name: count, dtype: int64

Os cursos foram escolhidos pois já existiam em 2002, tendo 20 anos de curso.

33 cursos da sede e 12 das extensões.

### CD_ORIGEM_ALUNO

Mais interessante para a análise descritiva

In [14]:
df['CD_ORIGEM_ALUNO'].nunique(dropna=False)

8

In [15]:
np.sort(df['CD_ORIGEM_ALUNO'].unique())

array([ 0,  1,  2,  3,  4,  5,  6, 15], dtype=int64)

In [16]:
df['CD_ORIGEM_ALUNO'].value_counts(dropna=False).sort_index()

CD_ORIGEM_ALUNO
0      7322
1     27404
2      6925
3     17061
4      9967
5      2394
6       147
15        1
Name: count, dtype: int64

1        MARINGA             
2        REGIAO DE MARINGA   
3        OUTRAS REGIOES PR   
4        ESTADO DE SP        
5        OUTROS ESTADOS      
6        OUTROS PAISES       

> Este campo entrou em desuso em 2020 e não é mais utilizado. 

In [17]:
df[df['DT_ANO_INGRES'] >= 2020].shape[0]

9421

In [18]:
df[df['DT_ANO_INGRES'] >= 2020].groupby('DT_ANO_INGRES')['CD_ORIGEM_ALUNO'].value_counts()

DT_ANO_INGRES  CD_ORIGEM_ALUNO
2020           1                  1423
               3                   706
               0                   631
               2                   330
               4                   212
               5                    70
               6                     3
2021           0                  2287
               1                   356
               3                   219
               2                    95
               4                    35
               5                    21
               6                     2
2022           0                  2704
               1                   164
               3                    75
               2                    46
               4                    29
               5                    12
               6                     1
Name: count, dtype: int64

Aparentemente não entrou em desuso

### CD_TURNO

In [19]:
df['CD_TURNO'].nunique(dropna=False)

6

In [20]:
df['CD_TURNO'].unique()

array(['N', 'V', 'M', 'I', 'W', 'Y'], dtype=object)

In [21]:
df['CD_TURNO'].value_counts(dropna=False).sort_index()

CD_TURNO
I    28177
M    10512
N    30343
V      843
W      349
Y      997
Name: count, dtype: int64

I        INTEGRAL                       
M        MATUTINO            
N        NOTURNO             
V        VESPERTINO          
W        VESPERTINO/NOTURNO  
Y        MANHA/TARDE/NOITE 

Curso de história, com código 5, por exemplo, tem apenas 1 aluno integral. Isso é erro do DAA.

### DE_NATURALIDADE

In [22]:
df['DE_NATURALIDADE'].nunique(dropna=False)

417

In [23]:
df['DE_NATURALIDADE'].value_counts(dropna=False)

DE_NATURALIDADE
BRASILEIRA                 26236
BRASILEIRO                 23030
NaN                        20511
MARINGA                      282
MARINGA-PR                   126
                           ...  
SAO JOSE DOS PINHAIS-PR        1
CAMBE\/PR                      1
COLIDER\/MT                    1
CRUZEIRO-SP                    1
JOSE BONIFACIO                 1
Name: count, Length: 417, dtype: int64

> Acho que deveria ser a cidade de nascimento. Acho que este também não é um bom campo, pois também está em desuso, a maioria dos alunos estão com esse campo nulo. O que se usa mais é DE_NAT_ESTRANG o qual é a cidade de nascimento quando a pessoa é estrangeira e o próprio usuário digita como quiser, pois o campo NM_CIDADE_NASC é populado por um selectbox conforme cadastro dos correios e não tem cidades fora do Brasil.

### DH_MATRICULA

Ano calendário

In [24]:
df['DH_MATRICULA'].nunique(dropna=False)

651

In [25]:
df['DH_MATRICULA'].value_counts(dropna=False)

DH_MATRICULA
2004-02-01    3313
2007-02-01    3212
2005-02-01    3209
2003-02-01    3141
2002-02-01    3039
              ... 
2017-04-24       1
2014-03-20       1
2012-02-10       1
2013-03-05       1
2012-04-16       1
Name: count, Length: 651, dtype: int64

Não consta em `campos-dados-alunos-Alison.xlsx`, mas deve ser data da matrícula.

Não é interessante.

### DH_SITUACAO

Ano calendário

In [26]:
df['DH_SITUACAO'].nunique(dropna=False)

3588

In [27]:
df['DH_SITUACAO'].value_counts(dropna=False)

DH_SITUACAO
2002-02-01    2514
2020-02-12    1607
2003-02-01    1512
2010-12-23    1357
2022-06-09    1292
              ... 
2012-08-15       1
2014-10-08       1
2014-08-28       1
2012-08-01       1
2022-07-09       1
Name: count, Length: 3588, dtype: int64

Data da última situação do aluno

A partir de 2007 estará mais confiável. Antes pode ter informação errada. Foi feita uma tratativa para pegar o DH_MATRICULA se o DH_SITUACAO estivesse vazio.


Um caso: aluno entrou em 2014 e estava com a situação em branco, ou seja, matriculado. Quando um aluno tranca e volta, o DH_SITUACAO vai estar com a data da volta, e não da matrícula.

Outro caso: aluno com data de matrícula 2009 e data de ingresso 2006, o errado nesse caso seria o DH_MATRICULA. Antigamente era tudo manual.



Sempre que DH_SITUAÇÃO for diferente do DH_MATRICULA estará certo.

In [28]:
df[df['DH_SITUACAO'] == df['DH_MATRICULA']]

ANO_OCORRENCIA     CD_ALUNO  CD_CURSO  CD_ORIGEM_ALUNO CD_TURNO  \
10                  0   1429259211         2                3        N   
14               2003   1285772432        23                1        N   
15               2004   1725435563        12                1        M   
31               2002   1538577190        13                1        N   
32               2005   1374661052        12                2        N   
...               ...          ...       ...              ...      ...   
71215               0  10264903272        12                0        M   
71217               0  14796067716        42                0        I   
71218               0  16902075967         8                0        M   
71219               0  16711090796        30                0        I   
71220               0  11135259520        19                0        I   

      DE_NATURALIDADE DH_MATRICULA DH_SITUACAO  DT_ANO_INGRES DT_NASC_ALUNO  \
10         BRASILEIRO   2020-01-17  2020-01-17           2020    1973-08-11   
14         BRASILEIRO   2002-02-01  2002-02-01           2002    1971-05-07   
15         BRASILEIRO   2003-02-01  2003-02-01           2003    1971-05-07   
31         BRASILEIRO   2002-02-01  2002-02-01           2002    1978-10-31   
32         BRASILEIRA   2002-02-01  2002-02-01           2002    1975-04-20   
...               ...          ...         ...            ...           ...   
71215             NaN   2022-07-18  2022-07-18           2022    1998-08-31   
71217             NaN   2022-07-29  2022-07-29           2022    1981-01-16   
71218             NaN   2022-11-24  2022-11-24           2022    2001-07-15   
71219             NaN   2022-11-25  2022-11-25           2022    1999-12-13   
71220             NaN   2022-11-25  2022-11-25           2022    2000-03-04   

       LT_ESTADO_CIVIL LT_MODALIDADE LT_NACIONALIDADE  LT_RACA  \
10                 4.0             P                B        2   
14                 1.0             P                B        0   
15                 1.0             P                B        0   
31                 1.0             P                B        0   
32                 1.0             P                B        0   
...                ...           ...              ...      ...   
71215              NaN             P                E        0   
71217              2.0             P                B        0   
71218              1.0             P                B        0   
71219              2.0             P                B        3   
71220              1.0             P                B        3   

       NM_CIDADE_AULA NM_CIDADE_EST2GRAU NM_CIDADE_FERIAS    NM_CIDADE_NASC  \
10            MARINGA           CIANORTE          MARINGA          CIANORTE   
14            MARINGA                NaN              NaN               NaN   
15            MARINGA                NaN              NaN               NaN   
31           PAICANDU                NaN              NaN               NaN   
32     DOUTOR CAMARGO                NaN              NaN               NaN   
...               ...                ...              ...               ...   
71215             NaN             MALABO              NaN       BIOKO NORTE   
71217       APUCARANA      FOZ DO IGUACU        APUCARANA          CURITIBA   
71218         MARINGA       SANTA AMELIA              NaN       JACAREZINHO   
71219             NaN           TAMARANA              NaN          LONDRINA   
71220             NaN           TAMARANA              NaN  CANDIDO DE ABREU   

          NM_PAIS_NASC NU_SEMESTRE2_VEST SG_SEXO_ALUNO SG_UF_AULA  \
10              BRASIL                 I             M         PR   
14              BRASIL                 I             M         PR   
15              BRASIL                 I             M         PR   
31              BRASIL                 V             M         PR   
32              BRASIL                 V             F         PR   
...          

### DT_ANO_INGRES

Ano letivo

In [29]:
df['DT_ANO_INGRES'].nunique(dropna=False)

21

In [30]:
np.sort(df['DT_ANO_INGRES'].unique())

array([2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012,
       2013, 2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022],
      dtype=int64)

In [31]:
df['DT_ANO_INGRES'].value_counts(dropna=False).sort_index()

DT_ANO_INGRES
2002    3220
2003    3278
2004    3326
2005    3222
2006    3307
2007    3339
2008    4307
2009    4159
2010    3656
2011    3671
2012    3448
2013    3629
2014    3431
2015    3309
2016    2977
2017    3084
2018    3207
2019    3230
2020    3375
2021    3015
2022    3031
Name: count, dtype: int64

### DT_NASC_ALUNO

In [32]:
df['DT_NASC_ALUNO'].nunique(dropna=False)

11681

In [33]:
df['DT_NASC_ALUNO'].value_counts(dropna=False)

DT_NASC_ALUNO
0001-01-01    62
1991-03-01    29
1990-10-10    28
1990-02-08    24
1994-06-22    23
              ..
1965-01-31     1
1969-03-31     1
1972-01-06     1
1979-12-18     1
1969-01-25     1
Name: count, Length: 11681, dtype: int64

### LT_ESTADO_CIVIL

In [34]:
df['LT_ESTADO_CIVIL'].nunique(dropna=False)

8

In [35]:
df['LT_ESTADO_CIVIL'].unique()

array([ 1.,  2.,  4.,  3.,  0.,  6.,  5., nan])

In [36]:
df['LT_ESTADO_CIVIL'].value_counts(dropna=False).sort_index()

LT_ESTADO_CIVIL
0.0     1204
1.0    64671
2.0     3674
3.0     1001
4.0      278
5.0       15
6.0      146
NaN      232
Name: count, dtype: int64

0 ou em branco = NÃO INFORMADO

1 = SOLTEIRO(A)

2 = CASADO(A)

3 = SEPARADO(A)

4 = DIVORCIADO(A)

5 = VIÚVO(A)

6 = OUTROS

### LT_MODALIDADE

In [37]:
df['LT_MODALIDADE'].nunique(dropna=False)

1

In [38]:
df['LT_MODALIDADE'].value_counts(dropna=False).sort_index()

LT_MODALIDADE
P    71221
Name: count, dtype: int64

São todos alunos do presencial.

### LT_NACIONALIDADE

In [39]:
df['LT_NACIONALIDADE'].nunique(dropna=False)

5

In [40]:
df['LT_NACIONALIDADE'].unique()

array(['B', 'N', 'E', nan, 'R'], dtype=object)

In [41]:
df['LT_NACIONALIDADE'].value_counts(dropna=False)

LT_NACIONALIDADE
B      70395
NaN      297
R        216
N        179
E        134
Name: count, dtype: int64

B= BRASILEIRA 

E= ESTRANGEIRA

N= BRASILEIRO NATURALIZADO

R= BRASILEIRO NASCIDO NO EXTERIOR

Nulo ou em branco = NÃO INFORMADO

### LT_RACA

In [42]:
df['LT_RACA'].nunique(dropna=False)

6

In [43]:
np.sort(df['LT_RACA'].unique())

array([0, 1, 2, 3, 4, 5], dtype=int64)

In [44]:
df['LT_RACA'].value_counts(dropna=False).sort_index()

LT_RACA
0    20818
1     3171
2    37476
3      177
4     1526
5     8053
Name: count, dtype: int64

0 = NÃO DECLARADA

1 = AMARELO(A), DE ORIGEM ORIENTAL

2 = BRANCO(A)

3 = INDÍGENA OU DE ORIGEM INDÍGENA

4 = NEGRO(A)

5 = PARDO(A)/MUDALTO(A)

Nulo ou em branco = NÃO INFORMADO

### NM_CIDADE_AULA

In [45]:
df['NM_CIDADE_AULA'].nunique(dropna=False)

1267

In [46]:
df['NM_CIDADE_AULA'].value_counts(dropna=False)

NM_CIDADE_AULA
MARINGA               40691
CIANORTE               2706
UMUARAMA               2104
SARANDI                2028
MARIALVA               1343
                      ...  
VILHENA                   1
ITOBI                     1
GUA\U00CDRA               1
S. JORGE DO PATROC        1
REALEZA                   1
Name: count, Length: 1267, dtype: int64

> Nome da cidade de moradia durante o período de aulas

Acho estranho que tenha tantas cidade aula assim sendo que todos os alunos são presenciais

### NM_CIDADE_EST2GRAU

In [47]:
df['NM_CIDADE_EST2GRAU'].nunique(dropna=False)

1698

In [48]:
df['NM_CIDADE_EST2GRAU'].value_counts(dropna=False)

NM_CIDADE_EST2GRAU
MARINGA                           22860
NaN                               14661
CIANORTE                           2121
UMUARAMA                           1356
SARANDI                            1231
                                  ...  
TOLEDO\/CASCAVEL                      1
IRATI E CURITIBA                      1
COLEGIO DR.AIMONE SALERNO -TAQ        1
ITAGUAJA PR                           1
NOVA LARANJEIRAS                      1
Name: count, Length: 1698, dtype: int64

### NM_CIDADE_FERIAS

In [49]:
df['NM_CIDADE_FERIAS'].nunique(dropna=False)

1759

In [50]:
df['NM_CIDADE_FERIAS'].value_counts(dropna=False)

NM_CIDADE_FERIAS
MARINGA                29407
NaN                     2989
CIANORTE                2411
SARANDI                 1989
UMUARAMA                1617
                       ...  
PRES. EPIT\U00C1CIO        1
SAT ISABEL DO IVAI         1
OCAUCU                     1
SAO J. DOS CAMPOS          1
ARAGUAINA                  1
Name: count, Length: 1759, dtype: int64

### NM_CIDADE_NASC

In [51]:
df['NM_CIDADE_NASC'].nunique(dropna=False)

1502

In [52]:
df['NM_CIDADE_NASC'].value_counts(dropna=False)

NM_CIDADE_NASC
MARINGA           17790
NaN               13264
CIANORTE           1789
SAO PAULO          1703
UMUARAMA           1455
                  ...  
LIDIANOPOLIS          1
FARROUPILHA           1
PRES. EPITACIO        1
LIMA                  1
BRAGANTINA            1
Name: count, Length: 1502, dtype: int64

### NM_PAIS_NASC

In [53]:
df['NM_PAIS_NASC'].nunique(dropna=False)

53

In [54]:
df['NM_PAIS_NASC'].value_counts(dropna=False)

NM_PAIS_NASC
BRASIL             68872
NaN                 1751
JAPAO                342
ESTADOS UNIDOS        28
PARAGUAI              25
PORTUGAL              20
GUINE-BISSAU          17
CHINA                 15
ESPANHA               12
ARGENTINA             11
BOLIVIA               11
REPUBLICA DO HA       10
ALEMANHA               9
CHILE                  8
AFEGANISTAO            8
ITALIA                 7
PERU                   7
VENEZUELA              7
ANGOLA                 6
FRANCA                 6
CABO VERDE             5
COLOMBIA               4
TAIWAN                 3
CANADA                 2
BR                     2
INGLATERRA             2
HAITI                  2
S.TOME PRINCIPE        2
GUINE EQUATORIA        2
REP. DO HAITI          2
GANA                   1
GUINE BISSAU           1
BRASILEIRO             1
10\/11\/1999           1
REPUBLICA ARABE        1
COSTA DO MARFIM        1
BRAZIL                 1
27032001               1
FOZ DO IGUACU          1
MOCAMBIQUE  

### NU_SEMESTRE2_VEST

In [55]:
df['NU_SEMESTRE2_VEST'].nunique(dropna=False)

9

In [56]:
df['NU_SEMESTRE2_VEST'].unique()

array(['I', nan, 'V', '2', 'S', 'A', 'E', 'G', 'R'], dtype=object)

In [57]:
df['NU_SEMESTRE2_VEST'].value_counts(dropna=False).sort_index()

NU_SEMESTRE2_VEST
2          7
A       5781
E       1128
G         22
I      27573
R        360
S        373
V      31836
NaN     4141
Name: count, dtype: int64

É o processo seletivo que o candidato fez para se tornar aluno, e atualmente usamos esses valores:

I = Vestibular de Inverno

V = Vestibular de Verão

A = PAS

D = Vestibular EAD

R = Vagas Remanescentes para Candidatos do Vestibular

E = Vagas Remanescentes para quem fez ENEM

G = Vagas Remanescentes para Refugiados ou Imigrantes em Situação de Vulnerabilidade 

S = SiSU

### SG_SEXO_ALUNO

In [58]:
df['SG_SEXO_ALUNO'].nunique(dropna=False)

2

In [59]:
df['SG_SEXO_ALUNO'].value_counts(dropna=False)

SG_SEXO_ALUNO
F    37487
M    33734
Name: count, dtype: int64

### SG_UF_AULA

In [60]:
df['SG_UF_AULA'].nunique(dropna=False)

31

In [61]:
df['SG_UF_AULA'].unique()

array(['PR', 'SP', nan, 'SC', 'GO', 'PA', 'MT', '--', 'S', 'RS', 'AC',
       'MS', 'RJ', 'MG', 'DF', 'P', 'SO', 'RO', 'TO', 'BA', 'RR', 'RN',
       'AL', 'PE', 'ES', 'AM', 'CE', 'MA', 'PB', 'PI', 'AP'], dtype=object)

In [62]:
df['SG_UF_AULA'].value_counts(dropna=False)

SG_UF_AULA
PR     64121
SP      4965
NaN     1378
MS       312
MG        93
MT        90
SC        85
AC        37
GO        25
RS        18
RJ        17
RO        16
BA        13
DF        12
--         6
TO         5
PA         4
MA         4
S          3
ES         3
CE         2
RR         2
RN         2
PI         1
PB         1
P          1
AM         1
PE         1
AL         1
SO         1
AP         1
Name: count, dtype: int64

Estado de moradia durante o período de aulas

Não faz sentido ser o estado de moradia no período de aulas

### SG_UF_EST2GRAU

In [63]:
df['SG_UF_EST2GRAU'].nunique(dropna=False)

33

In [64]:
df['SG_UF_EST2GRAU'].unique()

array(['PR', nan, 'SP', 'MS', 'RJ', 'MT', 'SC', 'AC', 'RO', 'ES', 'AP',
       'BA', 'MG', 'PI', 'RS', 'GO', 'PA', 'MA', 'PE', 'TO', 'DF', 'CE',
       'SE', 'PB', 'AM', 'AL', 'RN', 'EX', 'RR', 'US', '3', '0', 'CH'],
      dtype=object)

In [65]:
df['SG_UF_EST2GRAU'].value_counts(dropna=False)

SG_UF_EST2GRAU
PR     47135
NaN    14650
SP      7543
MS       564
MT       293
SC       185
MG       180
RO       100
RJ        85
GO        77
RS        69
BA        50
PA        49
TO        31
DF        27
CE        27
EX        25
ES        17
MA        14
AM        14
PE        13
AP        12
AL        11
PB        10
AC         9
RR         7
PI         7
SE         7
RN         4
0          3
US         1
3          1
CH         1
Name: count, dtype: int64

Estado do estabelecimento de ensino em que fez o 2º grau (ensino médio). Se for exterior será EX.

### SG_UF_FERIAS

In [66]:
df['SG_UF_FERIAS'].nunique(dropna=False)

32

In [67]:
df['SG_UF_FERIAS'].unique()

array(['PR', nan, 'SP', 'MG', 'SC', 'MT', 'GO', 'AC', 'MS', 'RS', 'RO',
       '--', 'S', 'PA', 'P', 'BA', 'PB', 'RJ', 'ES', 'TO', 'DF', 'RN',
       'CE', 'SE', 'MA', 'PE', 'RR', 'R', 'AL', 'PI', 'AM', 'AP'],
      dtype=object)

In [68]:
df['SG_UF_FERIAS'].value_counts(dropna=False)

SG_UF_FERIAS
PR     56591
SP      9339
NaN     3443
MS       669
MT       295
MG       238
SC       170
GO        89
RO        88
AC        63
RS        46
RJ        42
BA        40
DF        21
PA        13
TO        12
ES        11
CE         7
MA         6
--         6
P          5
RN         5
PI         4
PB         4
PE         3
AM         3
RR         2
S          2
SE         1
R          1
AL         1
AP         1
Name: count, dtype: int64

Estado de moradia durante o período de férias

### SG_UF_NASC_ALUNO

In [69]:
df['SG_UF_NASC_ALUNO'].nunique(dropna=False)

32

In [70]:
df['SG_UF_NASC_ALUNO'].unique()

array(['PR', 'SP', 'RS', 'PB', 'MS', 'RJ', 'SC', 'MG', nan, 'MT', 'GO',
       'RO', 'DF', 'PA', 'PE', 'MA', 'BA', 'AL', 'AM', 'PI', 'ES', 'AC',
       '5', 'CE', 'SE', 'AP', 'RN', '1', 'WP', 'TO', 'PY', 'RR'],
      dtype=object)

In [71]:
df['SG_UF_NASC_ALUNO'].value_counts(dropna=False)

SG_UF_NASC_ALUNO
PR     51503
SP     13942
MS      1071
NaN      962
MT       656
MG       634
SC       453
RO       365
RS       332
RJ       324
GO       187
BA       138
PA       126
DF        88
PE        74
CE        65
AM        49
PB        40
ES        37
MA        31
AL        25
RN        25
TO        23
AP        17
PI        15
AC        15
SE        14
RR         6
WP         1
1          1
PY         1
5          1
Name: count, dtype: int64

Estado de nascimento

### ST_ALUNO

In [72]:
df['ST_ALUNO'].nunique(dropna=False)

18

In [73]:
df['ST_ALUNO'].unique()

array(['2', '7', '8', '9', nan, 'J', 'F', '0', 'K', '1', 'P', 'O', '6',
       'I', '4', 'B', 'H', 'M'], dtype=object)

In [74]:
df['ST_ALUNO'].value_counts(dropna=False).sort_index()

ST_ALUNO
0       1873
1        111
2      10158
4          1
6        526
7      34016
8      13040
9       1244
B         52
F         28
H          7
I         48
J        447
K        568
M         17
O          2
P        310
NaN     8773
Name: count, dtype: int64

NaN - Matriculado(a) 

0 - **EVASÃO** - Matrícula Cancelada pela UEM 

1 - Trancado(a) 

2 - **EVASÃO** - Matrícula Cancelada (Pelo aluno)

4 - **EVASÃO** - Transferido de Curso (Outro campus)

6 - **EVASÃO** - Transferido de IES

7 - Formado(a)

8 - **EVASÃO** - Cancelado por Abandono

9 - Jubilado(a) 

B - Transferido de Campus

F - Permuta Campus (Mesmo curso)

H - **EVASÃO** - Permuta Campus (Outro curso)

I - Falecido(a) 

J - **EVASÃO** - Matrícula Cancelada (Reprovação por falta por dois anos consecutivos)

K - **EVASÃO** - Matrícula Cancelada (Reprovação por nota duas vezes consecutivas – Regimento antigo)

M - Em Mobilidade Estudantil

O - **EVASÃO** - ND - Situação lançada em meados de 2001 para alunos com ocorrência de “Transferência interna de curso”. Pode incluir outras situações

P - **EVASÃO** - Transferido de Curso (Mesmo campus)



separar formados, evasão e matriculados

filtrar falecidos

agrupar por cancelamento, transferencia, matriculado, formado, jubilado

In [75]:
evasao = ['0', '2', '4', '6', '8', 'P', 'O', 'K', 'J', 'H']
df[df['ST_ALUNO'].isin(evasao) == True].shape[0]

26932

### TP_DEFIC_INEP

In [76]:
df['TP_DEFIC_INEP'].nunique(dropna=False)

32

In [77]:
df['TP_DEFIC_INEP'].unique()

array(['NNNNNNNN', nan, 'NNNSNNNN', 'NNNNNNNNNNNNN', 'SNNNNNNN',
       'SNSSNNNN', 'NNSNNNNN', 'SNSNNNNN', 'SNNNNNNNNNNNN', 'NNSSNNNN',
       'NNNNNSNN', 'NNNNNNNNSSNNN', 'SNNSNNNN', 'NNNNNNNNSNNNN',
       'SSNNNNNN', 'NSNNNNNN', 'NNSNNNSN', 'SNNNSNNN', 'NNNSNNNNNNNNN',
       'NNNNNNSN', 'NNSNNNNNNNNNN', 'NNNNSNNN', 'NNSNNNSNNNNNN',
       'SSNNNNNNNNNNN', 'NNNNNNNNSNNSN', 'NNNNSNNNNNNNN', 'NNNNNNNNNNNSN',
       'NNNNNNNNNSNNN', 'SNNNNNNNNNNSN', 'NNNNNNNNSSNSN', 'NSNNNNNNNNNNN',
       'SNNNSSNN'], dtype=object)

In [89]:
df['TP_DEFIC_INEP'].value_counts(dropna=False)

TP_DEFIC_INEP
NNNNNNNN         49921
NaN              16099
NNNNNNNNNNNNN     4027
SNNNNNNN           873
NNNSNNNN            77
NNSNNNNN            63
SNNNNNNNNNNNN       55
NNNNNNNNNNNSN       13
NNNSNNNNNNNNN       13
NNNNSNNN            12
SNSNNNNN             9
SNNSNNNN             7
NNNNNNNNNSNNN        7
NNSNNNNNNNNNN        6
NNNNNNNNSNNNN        6
NSNNNNNN             6
NNNNSNNNNNNNN        4
NNNNNNNNSSNNN        3
SSNNNNNN             3
NNSNNNSNNNNNN        2
NNSNNNSN             2
SNNNSNNN             2
NNSSNNNN             2
NNNNNNSN             1
SSNNNNNNNNNNN        1
NNNNNNNNSNNSN        1
NNNNNSNN             1
SNSSNNNN             1
SNNNNNNNNNNSN        1
NNNNNNNNSSNSN        1
NSNNNNNNNNNNN        1
SNNNSSNN             1
Name: count, dtype: int64

"Deficiência do aluno conforme cadastro do INEP, e tem o seguinte formato:
NNNNNNNNNNNNN
sendo cada letra podendo ser S (sim) ou N (não), com cada posição significando:

1 Baixa Visão

2 Cegueira

3 Deficiência Auditiva

4 Deficiência Física

5 Deficiência Intelectual

6 Deficiência Multipla

7 Surdez

8 Surdocegueira

9 Autismo

10 Sindrome de Asperger

11 Sindrome de Rett

12 Altas Hbilidades/Superdotação

13 Trasntorno desintegrativo de Infrancia

Esse padrão de 13 posições começou a ser utilizado em 2020 em diante, e antes era utilizado apenas 8 posições, com os mesmos significados.

---

Esse campo começou a ser utilizado mais ou menos em 2010, e antes era utilizado o campo TP_DEFICIENCIA, o qual pode 
ter esses valores:

A  Auditiva

F  Física

M  Múltipla

N  Nenhuma

O  Outras

PC Paralisia Cerebral

VP  Visão Parcial

VT  Visão Total

Este campo TP_DEFICIENCIA mantêm-se atualizado mesmo após a criação do TP_DEFIC_INEP. Quando um aluno ou a DAA preenche o campo TP_DEFIC_INEP o sistema faz uma dedução de qual campo preencher em TP_DEFICIENCIA."

### TP_ESCOLA

In [78]:
df['TP_ESCOLA'].nunique(dropna=False)

10

In [79]:
df['TP_ESCOLA'].unique()

array(['GI', nan, 'E', 'SG', 'P', 'GP', 'SP', 'F', 'M', 'EX'],
      dtype=object)

In [80]:
df['TP_ESCOLA'].value_counts(dropna=False)

TP_ESCOLA
GI     33592
P      27342
E       4467
GP      3392
NaN     1154
SG       776
SP       266
F        107
EX        91
M         34
Name: count, dtype: int64

O tipo do estabelecimento de ensino em que fez o 2º grau (ensino médio), podendo ter os valores:

EX = Exterior

GI = Escola Pública Integralmente

GP = Escola Pública Parcialmente

P = Escola Privada Integralmente

SG = Supletivo Público

SP = Supletivo Particular

---

Há também os valore abaixo, os quais não são mais utilizados:

M = MUNICIPAL

E = ESTADUAL

F = FEDERAL

PE = EXAME DE PROFICIÊNCIA ENEM

### TP_INGRES_CURSO

In [81]:
df['TP_INGRES_CURSO'].nunique(dropna=False)

55

In [82]:
np.sort(df['TP_INGRES_CURSO'].unique())

array([ 1,  2,  3,  4,  5,  6,  7,  8, 10, 11, 12, 14, 15, 16, 17, 18, 19,
       21, 22, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38,
       39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 50, 53, 56, 57, 58, 59, 60,
       61, 62, 63, 98], dtype=int64)

Muitos valores, verificar documento

In [83]:
df[df['TP_INGRES_CURSO'] == 11]['NU_SEMESTRE2_VEST'].value_counts(dropna=False)

NU_SEMESTRE2_VEST
I      8
V      5
NaN    3
Name: count, dtype: int64

4141 nulos

6 - portador de diploma (1243 nulos)

32 - nova habilitacao (1113 nulos)

### TP_OPCAO_COTAS

In [84]:
df['TP_OPCAO_COTAS'].nunique(dropna=False)

6

In [85]:
df['TP_OPCAO_COTAS'].unique()

array(['N', nan, 'S', 'G', 'F', 'O'], dtype=object)

In [86]:
df['TP_OPCAO_COTAS'].value_counts(dropna=False)

TP_OPCAO_COTAS
N      62808
S       5862
NaN     2277
O        127
G        126
F         21
Name: count, dtype: int64

"Qual cota o aluno utilizou para entrar no curso. 
Não é necessariamente a cota que ele concorreu vaga no vestibular, pois há alguns casos que o aluno optou por alguma cota, porém por atingir nota ou por opção própria ele acaba entrando pela Universal.
Pode ter os seguintes valores:

N – Universal (sem cota)

S – Cotas Sociais

O – Cotas Sociais para Negros

G – Cotas para Negros

F – Cotas para PcD


D – Cotas para Docentes de Escola Pública

Em branco ou nulo: nenhuma cota ou não informado"

### Colunas

In [87]:
df.columns

Index(['ANO_OCORRENCIA', 'CD_ALUNO', 'CD_CURSO', 'CD_ORIGEM_ALUNO', 'CD_TURNO',
       'DE_NATURALIDADE', 'DH_MATRICULA', 'DH_SITUACAO', 'DT_ANO_INGRES',
       'DT_NASC_ALUNO', 'LT_ESTADO_CIVIL', 'LT_MODALIDADE', 'LT_NACIONALIDADE',
       'LT_RACA', 'NM_CIDADE_AULA', 'NM_CIDADE_EST2GRAU', 'NM_CIDADE_FERIAS',
       'NM_CIDADE_NASC', 'NM_PAIS_NASC', 'NU_SEMESTRE2_VEST', 'SG_SEXO_ALUNO',
       'SG_UF_AULA', 'SG_UF_EST2GRAU', 'SG_UF_FERIAS', 'SG_UF_NASC_ALUNO',
       'ST_ALUNO', 'TP_DEFIC_INEP', 'TP_ESCOLA', 'TP_INGRES_CURSO',
       'TP_OPCAO_COTAS'],
      dtype='object')